target/linux/realtek/files-5.10/drivers/clk/realtek/clk-rtl83xx.c

```diff
@@ -238,9 +239,9 @@ struct rtcl_clk {
 };
 
 static const struct rtcl_clk_info rtcl_clk_info[CLK_COUNT] = {
-	RTCL_CLK_INFO(CLK_CPU, "cpu_clk", "xtal_clk", "CPU"),
-	RTCL_CLK_INFO(CLK_MEM, "mem_clk", "xtal_clk", "MEM"),
-	RTCL_CLK_INFO(CLK_LXB, "lxb_clk", "xtal_clk", "LXB")
+	RTCL_CLK_INFO(CLK_CPU, "cpu_clk", "xtal_clk", "xtal_clk", "CPU"),
+	RTCL_CLK_INFO(CLK_MEM, "mem_clk", "xtal_clk", "xtal_clk", "MEM"),
+	RTCL_CLK_INFO(CLK_LXB, "lxb_clk", "xtal_clk", "xtal_clk", "LXB")
```

Comment:
Why have two identical names?

This patch doesn't seem to line up with the commit message either... If a clock may not necessarily have a fixed clock as source, why not just use the framework by putting everything in the DTS?